extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -36,134 +33,170 @@ public static function onBeforePageDisplay() {
+			$args[ 'mostrecent' ] !== 'false'
 		) {
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
+				'mostrecent',
+				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain()
+			)->plain();
 
 			return false;
 		}
 
 		// size must be integer in range
-		if ( isset( $args['size'] ) ) {
-			$size = $args['size'];
+		if ( isset( $args[ 'size' ] ) ) {
+			$sizeArg = $args[ 'size' ];
+			$size = ctype_digit( $sizeArg ) ? intval( $sizeArg ) : $sizeArg;
 
-			if ( !ctype_digit( $size ) ||
-				intval( $size ) > self::ITEMS_MAX ||
-				intval( $size ) < self::ITEMS_MIN
+			if ( !is_int ( $size ) ||
+				$size > self::ITEMS_MAX ||
+				$size < self::ITEMS_MIN
 			) {
-				// TODO: Refactor error messages to avoid direct concatenation, see JPN-657
-				$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'size' )->plain() .
-					wfMessage( 'embeddable-discussions-parameter-error-range',
-						self::ITEMS_MIN , self::ITEMS_MAX )->plain();
+				$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
+					'size',
+					wfMessage( 'embeddable-discussions-parameter-error-range', self::ITEMS_MIN, self::ITEMS_MAX )->plain()
+				)->plain();
 
 				return false;
 			}
 		}
 
 		// columns must be integer in range
-		if ( isset( $args['columns'] ) ) {
-			$columns = $args['columns'];
+		if ( isset( $args[ 'columns' ] ) ) {
+			$columnsArg = $args[ 'columns' ];
+			$columns = ctype_digit( $columnsArg ) ? intval( $columnsArg ) : $columnsArg;
 
-			if ( !ctype_digit( $columns ) ||
-				intval( $columns ) > self::COLUMNS_MAX ||
-				intval( $columns ) < self::COLUMNS_MIN
+			if ( !is_int( $columns ) ||
+				$columns > self::COLUMNS_MAX ||
+				$columns < self::COLUMNS_MIN
 			) {
-				// TODO: Refactor error messages to avoid direct concatenation, see JPN-657
-				$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'columns' )->plain() .
-					wfMessage( 'embeddable-discussions-parameter-error-range', self::COLUMNS_MIN , self::COLUMNS_MAX )->plain();
+				$errorMessage = wfMessage( 'embeddable-discussions-parameter-error',
+					'columns',
+					wfMessage( 'embeddable-discussions-parameter-error-range', self::COLUMNS_MIN, self::COLUMNS_MAX )->plain()
+				)->plain();
 
 				return false;
 			}
 		}
 
 		// category must be a valid category
-		if ( $modelData['invalidCategory'] ) {
-			// TODO: Refactor error messages to avoid direct concatenation, see JPN-657
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', $args['category'] )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-category' )->plain();
+		if ( !empty( $modelData[ 'invalidCategory' ] ) ) {
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', $args[ 'category' ],
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "category" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)